### PR TITLE
export types

### PR DESCRIPTION
**Pull request by samiramkr**
## Hey, I just made a Pull Request!

I'm adding the Kubernetes plugin in BIN and I found out some types is needed to be exported from BOSS.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey @samiramkr

I think we should add a new file at `./kubernetes-auth-provider/index.ts` which has the following:

```ts
export { kubernetesAuthProvidersApiRef } from './types';
export { KubernetesAuthProviders } from './KubernetesAuthProviders';
```

and then change the `plugins/kubernetes/src/index.ts` to have this:

```ts
export * from './kubernetes-auth-provider';`
```

File: .changeset/large-penguins-yell.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kubernetes': patch
+'@backstage/plugin-kubernetes-backend': patch
```
Comment (benjdlambert) [inline .changeset/large-penguins-yell.md:3]: Don't need this package bumped here :)

Comment (samiramkr) [reply .changeset/large-penguins-yell.md:3]: sure :)

Comment (benjdlambert) [reply .changeset/large-penguins-yell.md]: Sorry @samiramkr you'll still need to bump `'@backstage/plugin-kubernetes': patch` just not `kuberenetes-backend` :_

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**